KIPs/kip-113.md

```diff
@@ -0,0 +1,216 @@
+
+- Register a public key and a proof-of-possession
+- Unregister a public key
+- View the registered public keys
+
+#### Methods
+
+##### registerPublicKey
+
+Registers the given public key associated with the `msg.sender` address.
+
+The function validates the following requirements:
+
+- The function MUST revert if `publicKey.length != 48` or `pop.length != 96`
+- The function MUST revert if `publicKey` or `pop` is equal to zero
+- The function SHOULD authenticate and authorize `msg.sender`
+
+The function emits a `PubkeyRegistered` event.
+
+_Note_ The function is not able to verify the validity of the public key and the proof-of-possession due to the lack of [EIP-2537](https://eips.ethereum.org/EIPS/eip-2537). See [validation](#validation) for off-chain validation.
+
+```solidity
+function registerPublicKey(bytes memory publicKey, bytes memory pop)
+```
+
+##### unregisterPublicKey
+
+Unregisters the public key associated with the given `cnNodeId` address.
+
+- The function validates the following requirements:
+- The function MUST revert if `cnNodeId` has not been registered
+- The function SHOULD authenticate and authorize `msg.sender`
+
+The function emits a `PubkeyUnregistered` event.
+
+```solidity
+function unregisterPublicKey(address cnNodeId)
+```
+
+##### getInfo
+
+Returns the public key and proof-of-possession given a `cnNodeId`.
+
+The function MUST revert if given `cnNodeId` is not registered.
+
+_Note_ The function is not able to verify the validity of the public key and the proof-of-possession due to the lack of [EIP-2537](https://eips.ethereum.org/EIPS/eip-2537). See [validation](#validation) for off-chain validation.
+
+```solidity
+function getInfo(address cnNodeId) external view returns (BlsPublicKeyInfo pubkey)
+```
+
+##### getAllInfo
+
+Returns all the stored node IDs, public keys, and proof-of-possessions at once.
+
+_Note_ The function is not able to verify the validity of the public key and the proof-of-possession due to the lack of [EIP-2537](https://eips.ethereum.org/EIPS/eip-2537). See [validation](#validation) for off-chain validation.
+
+```solidity
+function getAllInfo() external view returns (address[] nodeId, BlsPublicKeyInfo[])
+```
+
+### Core Logic Overview
+
+A Klaytn node must specify the target block number; starting from the target block number, the node shall read the contract and use its data. Before the target block number, the contract should be ready to use, meaning that deploying Registry contract and registering the public keys should be done.
+
+#### Chain Configuration
+
+In the ChainConfig, the following fields are introduced. All node operators in a network must update `genesis.json` configuration with the same value. The configuration values for Baobab and Cypress networks are hard-coded on the client source code.
+
+- kip113CompatibleBlock: the block number which Klaytn node starts using Registry
+- kip113ContractAddress: the address of Registry
```

Comment:
```suggestion
- kip113ContractAddress: the address of a contract that implements IKIP_113.
```

Reply:
should i also update "Registry" in `kip113CompatibleBlock`?

Reply:
@ian0371 It would be also good if you agree with the change.

Reply:
let's leave it for brevity